docs/care/CEP/Draft/0010-labs.md

```diff
@@ -0,0 +1,370 @@
+        return f"Lab Order {self.lab_order.external_id} - Consultation {self.consultation.id}"
+
+
+class LabOrderSample(BaseModel):
+    """
+    Model to store the samples needed for the lab order. The sample can be in different states like Pending, Collected, In Transit, Received, Rejected, Processing, Under Review, Invalid, Completed, Cancelled.
+
+    The LabOrderSample instances can be created automatically based on the investigations ordered in the LabOrder, all the unique SampleType instances in the LabOrder.investigations can be created as LabOrderSample instances.
+    """
+
+    lab_order = models.ForeignKey(
+        LabOrder, on_delete=models.CASCADE, related_name="samples"
+    )
+    sample_type = models.ForeignKey(SampleType, on_delete=models.PROTECT)
+    status = models.CharField(
+        max_length=20,
+        choices=[
+            ("PENDING", "Pending"),
+            ("COLLECTED", "Sample Collected"),
+            ("IN_TRANSIT", "Sample sent to Lab"),
+            ("RECEIVED", "Sample received at Lab"),
+            ("REJECTED", "Sample Rejected"),
+            ("PROCESSING", "Test Ongoing"),
+            ("UNDER_REVIEW", "Result under review"),
+            ("INVALID", "Invalid Result"),
+            ("COMPLETED", "Test Completed"),
+            ("CANCELLED", "Test Cancelled"),
+        ],
+        default="PENDING",
+    )
+    collected_by = models.ForeignKey(
+        User, on_delete=models.PROTECT, null=True, blank=True
+    )
+    collected_date = models.DateTimeField(null=True, blank=True)
+    barcode = models.CharField(max_length=100, unique=True, null=True, blank=True)
+
+    def __str__(self):
+        return f"{self.sample_type} for {self.lab_order}"
+
+
+class LabOrderInvestigation(BaseModel):
+    """
+    Model to store the investigations ordered in the lab order. The investigations can be in different states like Pending, In Progress, Completed, Cancelled.
+    """
+
+    lab_order = models.ForeignKey(
+        LabOrder, on_delete=models.CASCADE, related_name="investigations"
+    )
+    investigation = models.ForeignKey(Investigation, on_delete=models.CASCADE)
+    lab_order_sample = models.ForeignKey(
+        LabOrderSample, on_delete=models.CASCADE, related_name="investigations"
+    )
+    status = models.CharField(
+        max_length=20,
+        choices=[
+            ("PENDING", "Pending"),
+            ("IN_PROGRESS", "In Progress"),
+            ("COMPLETED", "Completed"),
+            ("CANCELLED", "Cancelled"),
+        ],
+        default="PENDING",
+    )
+    notes = models.TextField(blank=True, null=True)
+
+    class Meta:
+        unique_together = ("lab_order", "investigation")
+
+    def __str__(self):
+        return f"{self.investigation.name} for {self.lab_order}"
+
+
+class InvestigationResult(BaseModel):
+    """
+    Model to store the results for the investigations. The results can be in different states like Received, Under Review, Invalid, Accepted.
+    """
+
```

Comment:
Add a FK to `LarOrder` also. Helps in the reporting part.

Reply:
Sure, like reverse relations right?